org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/CodeTemplatePreferences.java

```diff
@@ -191,4 +196,47 @@
 	 */
 	public static final String CODETEMPLATE_RECORDSNIPPET_PUBLIC = "${package_header}/**\n * ${type_name}\n */\npublic record ${type_name}(${cursor}) {\n}";
 
+	public static class Month extends SimpleTemplateVariableResolver {
+		public Month() {
+			super("month", "Current month"); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+
+		@Override
+		protected String resolve(TemplateContext context) {
+			return Integer.toString(Calendar.getInstance().get(Calendar.MONTH));
+		}
+	}
+
+	public static class Day extends SimpleTemplateVariableResolver {
+		public Day() {
+			super("day", "Current day of month"); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+
+		@Override
+		protected String resolve(TemplateContext context) {
+			return Integer.toString(Calendar.getInstance().get(Calendar.DAY_OF_WEEK));
```

Comment:
=> DAY_OF_MONTH

Reply:
Also need to fix the js code https://github.com/redhat-developer/vscode-java/blob/master/src/fileEventHandler.ts#L73-L74

Reply:
Right. With the fixes, creating a comment and delegating to JDT-LS now gives :

/**
* typeComment rgrunber
* Aug 18, 2021 3:14:34 PM
* 2021 08 18 15 14
* Test.java  Test
 */

However, the client-side filled-in values on new file, give :

/**
* typeComment rgrunber
* 8/18/2021 3:13:09 PM
* 2021 07 03 15 13
* Test.java  Test
*/

So month, and day are clearly wrong. I'll fix these as part of https://github.com/redhat-developer/vscode-java/issues/1987 since I was going to add a shortmonth there anyways.